src/Control/Biegunka/Source/Git.hs

```diff
@@ -97,29 +93,43 @@ git_ :: URI -> FilePath -> Script 'Sources ()
 git_ u p = git' u p mempty
 
 updateGit :: URI -> FilePath -> Git -> IO (Maybe String)
-updateGit u p Git { _remote, _branch } =
+updateGit u p Git { _branch, _failIfAhead } =
   doesDirectoryExist p >>= \case
     True -> do
-      let rbr = _remote </> _branch
+      let rbr = "origin" </> _branch
       before <- gitHash
+      remotes <- askGit p ["remote"]
+      when ("origin" `notElem` lines remotes) $
+        askGit' p ["remote", "add", "origin", u]
       -- Typical git: <https://stackoverflow.com/questions/11347712/git-fetch-only-for-current-branch>
-      _ <- askGit p ["fetch", _remote, _branch] -- TODO: track upstream
-      _ <- askGit p ["checkout", "-B", _branch, "--track", rbr] -- TODO: track upstream
-      _ <- askGit p ["merge", rbr, _branch] -- TODO: rebase
+      askGit' p ["fetch", "origin", _branch]
+      currentBranch <- lines `fmap` askGit p ["rev-parse", "--abbrev-ref", "HEAD"]
+      when (currentBranch /= [_branch]) $
```

Comment:
Right, `head` should not ever be used, but `currentBranch` being a list of strings does not make any sense either. How about
```
currentBranch <- fmap (listToMaybe . lines) (askGit p ["rev-parse", "--abbrev-ref", "HEAD"])
when (currentBranch /= Just _branch) $
  ...
```
?